### PR TITLE
Implement house price regression example workflow

### PR DESCRIPTION
**Pull request by chezou**
This example introduces regression workflow using Boston house pricing.

- [x] Add feature scaling with min-max
- [x] [Optional] Add hyper parameters such as regularization
- [ ] [Optional] Add feature selection

While we discussed scaling features is important for online learning, I didn't scale features because of the precision accuracy.

## Preprocessing conditions

- All features
- Top 4 correlated features: `lstat`, `rm`, `ptratio`, `indus`
- Top 4 features with min-max rescaling

| |All features| Top 4 features| Top 4 with min-max |
|---|---|---|---|
|**RMSE**|13.03|**7.83**|20.29|
|**MAE**|11.24|**6.18**|17.87|

### CODE REVIEW

**Comment by chezou** [comment]
@myui Could you have a look at this PR?

**Comment by chezou** [comment]
- [ ] Explore appropriate `eta0`
- [ ] Try with/without log scaling for features

**Comment by chezou** [comment]
With/without log1p scaling, eta0=0.5 should be good for this dataset. Will keep using log1p scaling for stability.

**Top 4 features, add_bias, iter=100**

| RMSE  | MAE   | eta0  | 
|-------|-------|-------| 
| 16.88 | 15.65 | 5     | 
| 10.85 | 9.77  | 1     | 
| **5.98**  | 4.73  | **0.5**   | 
| 7.13  | 5.64  | 0.1   | 
| 7.25  | 5.41  | 0.05  | 
| 13.39 | 10.34 | 0.01  | 
| 23.21 | 21.33 | 0.001 | 

**Top 4 features, add_bias, log1p scaling, iter=100**

| RMSE   | MAE    | eta0  | 
|--------|--------|-------| 
| 9.61   | 8.75   | 5     | 
| 127.42 | 121.36 | 1     | 
| **5.99**   | 4.43   | **0.5**   | 
| 10.38  | 7.55   | 0.1   | 
| 11.07  | 8.03   | 0.05  | 
| 20.37  | 18.16  | 0.01  | 
| 24.15  | 22.44  | 0.001 |

File: machine-learning/house_price/common/prepare_data.dig [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+  ## log1p rescaling
+  +training:
+    td>: ../queries/vectorize_log1p_features.sql
+    target_table: ${source}_${phase}
+    create_table: ${phase}
+  
+  +test:
+    td>: ../queries/vectorize_log1p_features.sql
+    phase: test
```
Comment (myui) [inline machine-learning/house_price/common/prepare_data.dig]: race condition on `phase` (?)
https://console.treasuredata.com/app/workflows/534349/sessions/3871847

File: machine-learning/house_price/common/prepare_data.dig [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  +training:
+    td>: ../queries/split_train.sql
+    engine: presto
+    create_table: ${source}_${phase}
+
+  +test:
+    td>: ../queries/split_test.sql
+    engine: presto
+    phase: test
+    create_table: ${source}_${phase}
```
Comment (myui) [inline machine-learning/house_price/common/prepare_data.dig]: `${source}_test`

File: machine-learning/house_price/common/prepare_data.dig [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+  td>: ../queries/shuffle.sql
+  create_table: ${source}_shuffled
+
++split:
+  _parallel: true
+
+  +training:
+    td>: ../queries/split_train.sql
+    engine: presto
+    create_table: ${source}_${phase}
```
Comment (myui) [inline machine-learning/house_price/common/prepare_data.dig]: ${source}_train

File: machine-learning/house_price/common/prepare_data.dig [hunk 4]
```diff
@@ -0,0 +1,129 @@
+_export:
+  phase: training
```
Comment (myui) [inline machine-learning/house_price/common/prepare_data.dig]: remove `phase: training` export

File: machine-learning/house_price/common/prepare_data.dig (hunk 1 above)
Comment (myui) [reply machine-learning/house_price/common/prepare_data.dig]: ^ it was due to typo in my testing.

**Comment by myui** [comment]
It seems Hivemall's default regularization hyperparameter does too aggressive regularization for this dataset.

`-reg no` improves accuracy.
https://console.treasuredata.com/app/workflows/534499/sessions/3872588

File: machine-learning/house_price/queries/split_test.sql [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  ,ptratio
+  ,b
+  ,lstat
+  ,medv
+  ,rnd
+FROM
+  ${source}_shuffled
+WHERE
+  rnd > ${training_ratio}
+;
```
[the remark this answers is not in the document]
Comment (chezou) [reply machine-learning/house_price/queries/split_test.sql]: I guess adding it to shuffle.sql is better for random shuffling, isn't it?

**Comment by chezou** [comment]
After adding random shuffle, adding `-reg no` seems better than default reguralization method adagrad+rda.

| eta0  | adagrad -reg no | adagrad -reg rda | 
|-------|-----------------|------------------| 
| 0.001 | 23.92           | 24.21            | 
| 0.01  | 14.55           | 20.58            | 
| 0.05  | 12.15           | 12.43            | 
| 0.1   | 11.92           | 12.02            | 
| 0.5   | 7.70            | 5.80             | 
| 1     | 6.22            | **5.32**             | 
| 5     | **5.17**            | 589.72           |

File: machine-learning/house_price/regression.dig [hunk 6]
```diff
@@ -0,0 +1,26 @@
+_export:
+  !include : config/params.yml
+  td:
+    apikey: ${apikey}
```
[the remark this answers is not in the document]
Comment (chezou) [reply machine-learning/house_price/regression.dig]: fixed 509402e

File: machine-learning/house_price/README.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+| ... | ...    | ... | ... | ... | ... | ...  | ... | ... | ... | ... | ... | ... | ... | 
+
+`medv`, Median value of owner-occupied homes in $1000's, is the target value for regression.
+
+## Workflow
+
+```sh
+$ ./data.sh # prepare sample dataset
+$ tf wf push regressor # push workflow to TD
+$ tf wf start regressor regression --session now -p apikey=${YOUR_TD_API_KEY}
```
[the remark this answers is not in the document]
Comment (chezou) [reply machine-learning/house_price/README.md]: fixed 509402e

File: machine-learning/house_price/README.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
+# House pricing Prediction
+
+In this example, we predict house pricing using [Regression algorithm](http://hivemall.incubator.apache.org/userguide/regression/general.html).
```
[the remark this answers is not in the document]
Comment (chezou) [reply machine-learning/house_price/README.md]: fixed 4bc4d75

File: machine-learning/house_price/queries/split_test.sql (hunk 5 above)
[the remark this answers is not in the document]
Comment (chezou) [reply machine-learning/house_price/queries/split_test.sql]: fixed 80c91ae

File: machine-learning/house_price/queries/vectorize_log1p_features.sql [hunk 9]
```diff
@@ -0,0 +1,17 @@
+select
+  rowid
+  ,array_concat(
```
[the remark this answers is not in the document]
Comment (chezou) [reply machine-learning/house_price/queries/vectorize_log1p_features.sql]: fixed 4777020

File: machine-learning/house_price/common/prepare_data.dig (hunk 4 above)
Comment (chezou) [reply machine-learning/house_price/common/prepare_data.dig]: fixed daabb7d

File: machine-learning/house_price/common/prepare_data.dig (hunk 3 above)
Comment (chezou) [reply machine-learning/house_price/common/prepare_data.dig]: fixed daabb7d

File: machine-learning/house_price/common/prepare_data.dig (hunk 2 above)
Comment (chezou) [reply machine-learning/house_price/common/prepare_data.dig]: fixed daabb7d

**Comment by chezou** [comment]
All comments are done and use `-reg no` for better accuracy.

**Comment by myui** [comment]
LGTM 👍

**Pull request merged**